src/transformers/generation/utils.py

```diff
@@ -1280,6 +1281,20 @@ def _prepare_generation_config(
 
         return generation_config, model_kwargs
 
+    def _get_initial_cache_position(self, input_ids, model_kwargs):
+        past_length = 0
+        if "past_key_values" in model_kwargs:
+            if isinstance(model_kwargs["past_key_values"], Cache):
+                past_length = model_kwargs["past_key_values"].get_seq_length()
+            else:
+                past_length = model_kwargs["past_key_values"][0][0].shape[-2]
+        if "inputs_embeds" in model_kwargs:
+            cur_len = model_kwargs["inputs_embeds"].shape[1]
+        else:
+            cur_len = input_ids.shape[-1]
+        model_kwargs["cache_position"] = torch.arange(past_length, cur_len, device=input_ids.device)
```

Comment:
It was not working in case when past key values are passed into the "generate" in Jamba, because Jamba apparently makes attention mask shape based on cache position values. Tested all `generation` tests + LLama/Jamba/Gemma tests, all are passing on my end